### PR TITLE
feat: amp module detail and breadcrumb

### PR DESCRIPTION
**Pull request by sunzsolo**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Features
 
+- [#633](https://github.com/alleslabs/celatone-frontend/pull/633) feat: amp module detail and breadcrumb
```
Comment (songwongtp) [inline CHANGELOG.md:42]: remove `feat: `

File: src/lib/amplitude/track-event/trackInteraction.ts [hunk 2]
```diff
@@ -57,6 +57,8 @@ export const trackUseExpand = ({
     | "permission_address"
     | "event_box"
     | "unsupported_pool"
+    | "Module Function Accordian"
+    | "Module Struct Accordian"
```
Comment (songwongtp) [inline src/lib/amplitude/track-event/trackInteraction.ts:61]: use snake case e.g. `module_function...`

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/amplitude/track-event/trackInteraction.ts [hunk 3]
```diff
@@ -57,6 +57,8 @@ export const trackUseExpand = ({
     | "permission_address"
     | "event_box"
     | "unsupported_pool"
+    | "module_function_accordian"
```
Comment (evilpeach) [inline src/lib/amplitude/track-event/trackInteraction.ts:60]: accordion

File: src/lib/amplitude/track-event/trackInteraction.ts [hunk 4]
```diff
@@ -57,6 +57,8 @@ export const trackUseExpand = ({
...
+    | "module_struct_accordian"
```
Comment (evilpeach) [inline src/lib/amplitude/track-event/trackInteraction.ts:61]: ditto

File: src/lib/pages/module-details/index.tsx [hunk 5]
```diff
@@ -92,6 +92,14 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
     [moduleData.address, moduleData.moduleName, navigate, tab]
   );
 
+  const handleActionNavigate = useCallback(
+    (nextTab: TabIndex, fnType?: FunctionTypeTabs) => () => {
+      track(AmpEvent.USE_NAVIGATING_BUTTON, { Label: fnType ?? "History" });
```
Comment (evilpeach) [inline src/lib/pages/module-details/index.tsx:97]: `label`

**Review by evilpeach — APPROVED** [review]

File: src/lib/pages/module-details/index.tsx [hunk 6]
```diff
@@ -314,6 +344,10 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
                         historyCount={moduleHistoriesCount}
                         refetchCount={refetchHistoriesCount}
                         onViewMore={() => {
+                          // Will remove after revamp AMP structure to provided context for each page
+                          track(AmpEvent.USE_VIEW_MORE, {
+                            table: "Published Event",
```
Comment (songwongtp) [inline src/lib/pages/module-details/index.tsx:349]: `Module Published Events`

File: src/lib/pages/module-details/components/ModuleStruct.tsx [hunk 7]
```diff
@@ -51,6 +52,10 @@ export const ModuleStruct = ({ structs }: ModuleStructProps) => {
               />
             }
             onClick={() => {
+              trackUseExpandAll(
+                expandedIndexes.length ? "collapse" : "expand",
+                "[Module Detail] Struct"
```
Comment (songwongtp) [inline src/lib/pages/module-details/components/ModuleStruct.tsx:57]: `Module Struct`

File: src/lib/pages/module-details/components/ModuleStruct.tsx [hunk 8]
```diff
@@ -64,6 +69,7 @@ export const ModuleStruct = ({ structs }: ModuleStructProps) => {
             size="sm"
             rightIcon={<CustomIcon name="launch" />}
             onClick={() => {
+              trackUseViewJSON("Module Detail Struct");
```
Comment (songwongtp) [inline src/lib/pages/module-details/components/ModuleStruct.tsx:72]: `Module Struct`

File: src/lib/pages/module-details/index.tsx [hunk 9]
```diff
@@ -303,6 +329,10 @@ export const ModuleDetailsBody = ({ moduleData }: ModuleDetailsBodyProps) => {
                         txCount={moduleTxsCount}
                         refetchCount={refetchTxsCount}
                         onViewMore={() => {
+                          // Will remove after revamp AMP structure to provided context for each page
+                          track(AmpEvent.USE_VIEW_MORE, {
+                            table: "Module transactions",
```
Comment (songwongtp) [inline src/lib/pages/module-details/index.tsx:334]: `Module Transactions`

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**